pkg/utils/workloadidentity/jwt.go

```diff
@@ -34,6 +34,8 @@ type TokenIssuer interface {
 	// IssueToken generates JSON Web Token based on the provided subject, audiences, duration and claims.
 	// It returns the token and its expiration time if successfully generated
 	IssueToken(sub string, aud []string, duration int64, claims ...any) (string, *time.Time, error)
+	// IssuerURL returns the URL used as issuer value in the tokens.
+	IssuerURL() string
```

Comment:
I would rather just call it `Issuer` as there is no hard requirement that this is a URL.

```suggestion
	// Issuer is the authority that issues tokens. This value is populated in the "iss" claim of each token.
	Issuer() string
```